src/pages/index.js

```diff
@@ -95,5 +170,66 @@ const validateAddPlace = new FormValidator(config, addPlaceForm);
 validateAddPlace.enableValidation();
 
 const popupWithImage = new PopupWithImage({ popupSelector: "#image-modal" });
-
 popupWithImage.setEventListeners();
+
+// Avatar
+const updateAvatarModal = document.querySelector("#update-avatar");
+const avatarForm = updateAvatarModal.querySelector(".modal__form");
+const updateAvatarButton = document.querySelector(".profile__image_edit-icon");
+
+const validateUpdateAvatar = new FormValidator(config, avatarForm);
+validateUpdateAvatar.enableValidation();
+
+const popupUpdateAvatar = new PopupWithForm(
+  "#update-avatar",
+  handleUpdateAvatar
+);
+popupUpdateAvatar.setEventListeners();
+
+function handleUpdateAvatar({ link }) {
+  popupUpdateAvatar.showLoading();
+  return api
+    .updateAvatar(link)
+    .then((data) => {
+      userInfo.setAvatar(data.avatar);
+      popupUpdateAvatar.close();
+      avatarForm.reset();
+      validateUpdateAvatar.toggleButtonState();
+    })
+    .catch((err) => {
+      console.error(err);
+      alert(`${err}, something went wrong`);
+    })
+    .finally(() => {
+      popupUpdateAvatar.hideLoading();
+    });
+}
+
+updateAvatarButton.addEventListener("click", () => {
+  popupUpdateAvatar.open();
+});
+
+// Confirm delete Card
+export const popupConfirmDeleteCard = new PopupWithConfirmation(
+  "#confirm-modal",
+  (card) => handleDelete(card)
+);
+popupConfirmDeleteCard.setEventListeners();
+
+function handleDelete(card) {
+  popupConfirmDeleteCard.open(() => {
+    popupConfirmDeleteCard.showLoading();
+    api.deleteCard(card._cardData._id)
+      .then(() => {
+        card._handleDeleteCard();
```

Comment:
Needs correction: please rename this method to `deleteCard` as it's not an event handler and make it public as the private methods can't be used outside of a class